### ISSUE TITLE
Catalina upgrade -- node/terminal full disk access

### ISSUE THREAD

**Issue by stefangomez** [aws-amplify/amplify-cli#3153]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
Catalina issue with env pull/push

**Amplify CLI Version**
4.11.0

**To Reproduce**
Upgrade to Catalina, and Upgrade from Amplify 3.x to 4.x

**Expected behavior**
Amplify push work normally

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Mac OS Catalina
 - Node Version. 10, 12

**Additional context**
This is more informational *than* a bug report just in case others run into this issue. If you *upgrade* to catalina, and are also mid-upgrade of an amplify version that uses different configuration files, weird things start happening on amplify env pull --restore / amplify pull. I found others having similar issues with [watchman](https://github.com/facebook/watchman/issues/751#issuecomment-542300670)

Basically, if you're having issues where `amplify pull` or `amplify env pull --restore` create duplicate files with a <space> " " and a number after, it's because you need to allow terminal/node to have full disk access. I encountered issues because it created additional CustomResources files on a colleague's machine, and amplify would create CloudFormation Definitions for those resources with a <space> in them which is not allowed.

TLDR; if you're upgrading to Catalina, give terminal full disk access.

**Comment by kaustavghosh06** [comment]
@stefangomez Thanks for the information. This would be useful for anyone else trying to update to Catalina.

**Comment by 112Cracker** [comment]
Same situation occurred to me when I ran `amplify env pull --restore`, duplicated files with exactly the same situation like this popped out from nowhere. But my OS is mac Mojave, any ideas why this could happen? Thanks!

**Comment by ul** [comment]
I wonder why does it register the entire user directory as a Watchman root in the first place?
